examples/step-17/doc/intro.dox

```diff
@@ -13,58 +14,169 @@ within an MPI network, the resulting code will even be able to solve the
+necessary for the right hand side), or also some additional elements
+that make sure that every processor has access the solution components
+that live on the cells the processor owns (so-called 
+@ref GlossLocallyActiveDof "locally active DoFs") or also on neighboring cells
+(so-called @ref GlossLocallyRelevantDof "locally relevant DoFs").
+
+The interface the classes from the PETScWrapper namespace provide is very similar to that
 of the deal.II linear algebra classes, but instead of implementing this
 functionality themselves, they simply pass on to their corresponding PETSc
 functions. The wrappers are therefore only used to give PETSc a more modern,
 object oriented interface, and to make the use of PETSc and deal.II objects as
-interchangeable as possible.
-
-
-
-While the sequential PETSc wrappers classes do not have any advantage over
-their deal.II counterparts, the main point of using PETSc is that it can run
+interchangeable as possible. The main point of using PETSc is that it can run
 in %parallel. We will make use of this by partitioning the domain into as many
-blocks (``subdomains'') as there are processes in the MPI network. At the same
-time, PETSc provides dummy MPI stubs that allow to run the same program on a
-single machine if so desired, without any changes.
-
-
-
-Note, however, that the only data structures we parallelize are matrices and
-vectors. We do, in particular, not split up the <code>Triangulation</code> and
-<code>DoFHandler</code> classes: each process still has a complete copy of
+blocks ("subdomains") as there are processes in the MPI network. At the same
+time, PETSc also provides dummy MPI stubs, so you can run this program on a
+single machine if PETSc was configured without MPI.
+
+
+<h2>Parallelizing software with MPI</h2>
+
+Developing software to run in %parallel via MPI requires a bit of a change in
+mindset because one typically has to split up all data structures so that
+every processor only stores a piece of the entire problem. As a consequence,
+you can't typically access all components of a solution vector on each 
+processor -- each processor may simply not have enough memory to hold the
+entire solution vector. Because data is split up or "distributed" across
+processors, we call the programming model used by MPI "distributed memory
+computing" (as opposed to "shared memory computing", which would mean
+that multiple processors can all access all data within one memory
+space, for example whenever multiple cores in a single machine work
+on a common task). Some of the fundamentals of distributed memory
+computing are discussed in the 
+@ref distributed "Parallel computing with multiple processors using distributed memory"
+documentation module, which is itself a sub-module of the
+@ref Parallel "Parallel computing" module. 
+
+In general, to be truly able to scale to large numbers of processors, one
+needs to split <i>every</i> data structure within a program between the
```

Comment:
I would suggest to slightly reword this. We split every *global* data structure that is proportional to the size of the problem, but replicate local things such as FiniteElement, the local matrices and vectors, and so on. (One could imagine a case where those data structures are also split but it requires yet another mindset.)